src/ssl.c

```diff
@@ -153,6 +153,9 @@
 #ifdef OPENSSL_EXTRA
 /* Global pointer to constant BN on */
 static WOLFSSL_BIGNUM* bn_one = NULL;
+
+static const WOLFSSL_RAND_METHOD* rand_methods = NULL;
+static wolfSSL_Mutex rand_methods_mutex;
```

Comment:
Please wrap this in `#ifndef SINGLE_THREADED` otherwise you will get a build issue with `OPENSSL_EXTRA` and `SINGLE_THREADED`.

Reply:
Are you sure this is necessary?  I see other mutexes defined without this wrapper, and it builds fine for me without it with SINGLE_THREADED + OPENSSL_EXTRA.  It looks like wolfSSL_Mutex is defined as an int in `wc_port.h` and the mutex methods are overridden, is this not sufficient?